Tutorial on creating a bare metal application (#1305)
ARMmbed/Handbook: docs/bare_metal/mbed2_porting.md

## Validating the port

> To validate the bare metal target configuration, execute the Mbed OS Greentea test suite with the bare metal profile. This profile causes Greentea to skip a subset of the tests, either because the underlying functionality has not been ported to bare metal or because some tests require RTOS features (for examples, more complex tests based on multi-threading). All tests compatible with bare metal are enabled.

Reviewer: ```suggestion
To validate the bare metal target configuration, execute the Mbed OS Greentea test suite with the bare metal profile. This profile causes Greentea to skip a subset of the tests, either because the underlying functionality has not been ported to bare metal or because some tests require RTOS features (for examples, more complex tests based on multi-threading). It performs all the tests compatible with bare metal.
```

    <span class="tips">**Tip:** You can append `--compile` and fix build issues before running tests with `--run`.</span>

> 1. All tests should pass (or be automatically skipped), with exceptions when the target being ported is ultra-constrained (with 32KB of flash memory or less) in which case linking may fail for only _a few_ tests. For example,

Reviewer: ```suggestion
1. All tests should pass (or be automatically skipped), unless the target being ported is ultraconstrained (with 32KB or less of flash memory) - in which case linking may fail for _a few_ tests. For example:
```

    ARM toolchain:
    ```
    Error: L6407E: Sections of aggregate size 0x318 bytes could not fit into .ANY selector(s).
    ```

    GCC_ARM toolchain:
    ```
    region `FLASH' overflowed by 792 bytes
    ```

>     Please ignore tests with similar errors for now.

Reviewer: What does "for now" mean? Until later in the document? Later in the year?

Author: I'll remove "for now" to avoid confusion.
(If we have worked around those errors sometime in the future, we will then remove this line in my opinion.)
